test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -224,12 +248,10 @@ func setup(ctx context.Context, profile *performancev2.PerformanceProfile) func(
 	teardown := func(ctx2 context.Context) {
 		By(fmt.Sprintf("executing teardown - revert profile %q back to its intial state", profile.Name))
 		Expect(testclient.Client.Get(ctx2, client.ObjectKeyFromObject(initialProfile), profile))
-		profile.Spec = initialProfile.Spec
-		resourceVersion := profile.ResourceVersion
-		testprofiles.UpdateWithRetry(profile)
+		testprofiles.UpdateWithRetry(initialProfile)
 
 		// do not wait if nothing has changed
-		if resourceVersion != profile.ResourceVersion {
+		if initialProfile.ResourceVersion != profile.ResourceVersion {
```

Comment:
I understand this should always be true considering `if !updateNeeded`  at line 236, but I think we need to fetch the new PP again (instead initialProfile) before this comparison to compare the correct values.

Reply:
`profile` contains the most updated profile and `initialProfile` contains the profile values before the call to `Update()` if something has changed we expect  
 `initialProfile.ResourceVersion != profile.ResourceVersion` to be `true` 
But, if we'll fetch the latest profile into `initialProfile` then `initialProfile.ResourceVersion != profile.ResourceVersion` will *always* be false even if something has changed.

Reply:
right, I misread. thanks for elaborating